lib/MusicBrainz/Server/Controller/Role/Load.pm

```diff
@@ -107,6 +107,9 @@ role
         }
 
         if ($entity) {
+            if ($id_is_guid && $entity->gid ne $id) {
+                $c->res->redirect($c->uri_for($c->action, [$entity->gid], $c->req->params), 301);
```

Comment:
There's a small bug here involving actions with multiple captures. For example, in the path `/artist/b7ffd2af-418f-4be2-bdd1-22f8b48613da/alias/15504/edit` there are two captures: the gid and the alias id (15504). But when calculating the redirect here we only pass one capture arg (`[$entity->gid]`). Since for core entities the gid will always be the first capture arg, I believe we can get away with just making a copy of the old args and changing the first element:
```suggestion
                my @captures = @{ $c->req->captures };
                $captures[0] = $entity->gid;
                $c->res->redirect($c->uri_for($c->action, \@captures, $c->req->params), 301);
```
Mind fixing that and adding one more test case for /alias/id/edit?